frontend/lib/note_keeper.dart

```diff
@@ -26,15 +26,20 @@ class _NoteKeeperState extends ConsumerState<NoteKeeper> {
       return Container();
     }
 
-    return MaterialApp.router(
-      scaffoldMessengerKey: globalScaffold,
-      themeMode: ThemeMode.dark,
-      darkTheme: ThemeData.dark(),
-      routeInformationParser: const RoutemasterParser(),
-      routerDelegate: RoutemasterDelegate(routesBuilder: (context) {
-        final isAuthenticated = ref.watch(_isAuthenticateProvider);
-        return isAuthenticated ? routeLoggedIn : routeLoggedOut;
-      }),
+    return GestureDetector(
+      /// hide keyboard when tapping outside of the input fields.
+      onTap: () => FocusManager.instance.primaryFocus?.unfocus(),
+      child: MaterialApp.router(
+        debugShowCheckedModeBanner: false,
+        scaffoldMessengerKey: globalScaffold,
+        themeMode: ThemeMode.dark,
+        darkTheme: ThemeData.dark(),
+        routeInformationParser: const RoutemasterParser(),
+        routerDelegate: RoutemasterDelegate(routesBuilder: (context) {
+          final isAuthenticated = ref.watch(_isAuthenticateProvider);
```

Comment:
When we update the username, for example, the entire app reloads. It takes us from the profile page, shows the login view for a moment, then takes us back to the home page.

I was thinking we can somehow show a spinner loading and not really show that the entire app reloaded.

Reply:
For this, you can create a new issue and let's merge this PR.